api/services/MachineService.js

```diff
@@ -241,28 +241,56 @@ function increaseUsersMachineEndDate(user) {
  * @return {Promise}
  */
 function _createMachine() {
+  let machine = {};
+
+  // Wait 2min before checking if driver is not dummy (testing purpose)
+  const wait = (driverName() === 'dummy') ? (0) : (120000);
+  // Check machine's status every 10s if driver is not dummy (testing purpose)
+  const interval = (driverName() === 'dummy') ? (0) : (10000);
+
   return ConfigService.get('machinesName')
     .then((config) => {
-      const machine = _driver.createMachine({
+      return _driver.createMachine({
         name: config.machinesName
-      })
-        .then((machineCreated) => {
-          _createBrokerLog(machineCreated, 'Created');
-        })
-        .finally(() => {
-          const len = _awaitingMachines.length;
-          for (let i = 0; i < len; i++) {
-            if (_awaitingMachines[i] === machine) {
-              _awaitingMachines.splice(i, 1);
-            }
+      });
+    })
+    .then((res) => {
+      machine = res;
+      machine.status = 'booting';
+      _createBrokerLog(machine, 'Created');
+      return Machine.create(machine);
+    })
+    .finally(() => {
+      setTimeout(function (){
```

Comment:
This should be a `then`. We don't wait to check virtual machine's status if an error occurred on creation.